### DISCUSSION TITLE
Impossible to load model on visual studio code

### DISCUSSION THREAD

**Discussion by MaximeTut** [pycaret/pycaret#3042] [General]
Pycaret seems awesome in a notebook; but the time I win developing a model on a notebook I waste it to solve the number of bugs I have when I want to load a pipeline !!!

Why it is so complicated to load a pipeline I saved in a notebook ?

I use the same virtual environment between Notebook and VS Code, but whatever I do I have an error !! It is crazy.
Here are the errors :

ModuleNotFoundError: No module named 'toml'
ModuleNotFoundError: No module named 'imblearn'
Module ‘pycaret.internal.preprocess’ has no attribute ‘DataTypes_Auto_infer’

I am working with the streamlit lib. Is there a possibility of a conflict of dependencies ?

**Comment by ngupta23** [comment]
It is not possible to provide much guidance due to the lack of information in your original post. It is possible that the version of pycaret that you used to create the pkl is not the same as the version you are using to load the pkl file.  Also, it may be possible that the environment that you use to load the pkl is different from the one you used to create the model.

But these would be nothing more than a guess based on the info you have provided.

FYI... please review the guideline to submit a minimal reproducible example to assist with debugging: https://github.com/pycaret/pycaret/issues/new?assignees=&labels=bug&template=bug_report.yml&title=%5BBUG%5D%3A+

**Reply by MaximeTut** [reply]
Thanks for answering ^^
I created a virtual environement in conda. I trained a pipeline in a notebook, I saved the pipeline. I use VSCode with the same conda virtual environment, and if there is only pycaret in the virtual env I can load the pipeline in my python script. If I pip install streamlit, I can't do it anymore and I have the errors I wrote above. So I have the same venv and the same pycaret version.

In the notebook : 
```
import pandas as pd
from pycaret.regression import *
save_model(best_home_score, "best_home_score_pipeline")

```

In my python script : 

```
import streamlit as st
from pycaret.regression import load_model

df = pd.read_csv("df_featured.csv")

model_home_score = load_model('best_home_score_pipeline')
```

The last error I got when trying to pip install pycaret after pip install streamlit in the conda virtual environment: 

```
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

**Reply by ngupta23** [reply]
It looks like your error is somehow coming when you are installing a library (using pip). This should not come from `pycaret` as `pycaret` does not install any libraries when you run `load_model`.

**Reply by ngupta23** [reply]
What version of pycaret are you using by the way? Have you tried this using the latest pre-release version. You should not see any issues there.

```
pip install --pre pycaret
```

**Reply by MaximeTut** [reply]
This last error occured when I tried to install pycaret with pip after installing streamlit. All the errors I get is when both framework are installed, independently I have no error... I guess there are some conflitcs between them...

**Reply by ngupta23** [reply]
Did you try the pre release version? It works for me with streamlit.

**Reply by MaximeTut** [reply]
Oh damn, it works, I did'nt have any hope.. Many thanks ! :)
